lib/rmt/scc_sync.rb

```diff
@@ -9,62 +9,81 @@ def initialize(logger = nil)
 
   def sync
     @logger.info('Cleaning up the database')
-    Product.delete_all
-    Repository.delete_all
+    clean_up
 
     @logger.info('Downloading data from SCC')
-    api = SUSE::Connect::Api.new(Settings.scc.username, Settings.scc.password)
-    data = api.list_products
+    scc_api_client = SUSE::Connect::Api.new(Settings.scc.username, Settings.scc.password)
+    data = scc_api_client.list_products
 
     @logger.info('Updating the database')
     data.each do |item|
-      extensions = []
-      repositories = []
-
-      item[:extensions].each do |ext_item|
-        begin
-          extension = Product.find(ext_item[:id])
-        rescue
-          extension = Product.new
-          extension.attributes = ext_item.select { |k, _| extension.attributes.keys.member?(k.to_s) }
-          extension.save!
-        end
-        extensions << extension
-      end
+      @logger.debug("Adding product #{item[:name]}")
+      product = create_product(item)
+      create_service(item, product)
+    end
+
+    @logger.info('Done!')
+  end
+
+  protected
+
+  def clean_up
+    Product.delete_all
+    Repository.delete_all
```

Comment:
Nope, because they are linked to activations, it would be weird to delete and recreate them all the time. (Product IDs don't change, so it's fine for products)